tools/pythonpkg/tests/arrow/test_nested_arrow.py

```diff
@@ -88,9 +87,30 @@ def test_lists_roundtrip(self,duckdb_cursor):
 
         #LIST[LIST[LIST[LIST[LIST[INTEGER]]]]]]
         compare_results("SELECT list (lllle) llllle from (SELECT list (llle) lllle from (SELECT list(lle) llle from (SELECT LIST(le) lle FROM (SELECT LIST(i) le from range(100) tbl(i) group by i%10) as t) as t1) as t2) as t3")
-                
+
         compare_results('''SELECT grp,lst,cs FROM (select grp, lst, case when grp>1 then lst else list_value(null) end as cs
                         from (SELECT a%4 as grp, list(a) as lst FROM range(7) tbl(a) group by grp) as lst_tbl) as T;''')
+                #Tests for converting multiple lists to/from Arrow with NULL values and/or strings
+        compare_results("SELECT list(st) from (select i, case when i%10 then NULL else i::VARCHAR end as st from range(1000) tbl(i)) as t group by i%5")
+
+    def test_struct_roundtrip(self,duckdb_cursor):
+        if not can_run:
+            return
+        compare_results("SELECT a from (SELECT STRUCT_PACK(a := 42, b := 43) as a) as t")
```

Comment:
Could you add more tests with complex types:

* List of structs
* Lists of structs with lists
* These but with different types of lists internally (`struct(list, list, list)`, one is a large list, the other is a small list, the other is a fixed size list)
* Maps embedded in a struct
* List of maps
* Map with list as key and/or value
* Map with struct as key and/or value
* Struct that is NULL entirely (e.g. `case when a%2=0 then null else struct_pack(...)`)
* Map that is NULL entirely (same case statement)